app/models/user.rb

```diff
@@ -7,4 +8,28 @@ class User < ApplicationRecord
   validates :email, presence: true, length: { maximum: Settings.user.email.max_length }, format: { with: VALID_EMAIL_REGEX }, uniqueness: { case_sensitive: false }
 
   has_secure_password
+
+  def User.digest string
+    cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST :
+                                                  BCrypt::Engine.cost
+    BCrypt::Password.create string, cost: cost
+  end
+
+  def User.new_token
+    SecureRandom.urlsafe_base64
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update remember_digest: User.digest(remember_token)
+  end
+
+  def authenticated? remember_token
+    return false if remember_digest.nil?
+    BCrypt::Password.new(remember_digest).is_password? remember_token
+  end
+
+  def forget
+    update_attribute :remember_digest, nil
```

Comment:
dùng update

Reply:
nếu trong session không có thì nó sẽ tìm theo id người dùng gần nhất vừa log in anh ạ

Reply:
hiểu sai nhé tìm hiểu lại